Detectors/GlobalTracking/include/GlobalTracking/TrackCuts.h

```diff
@@ -0,0 +1,123 @@
+/// \brief Class to perform track cuts
+/// \author amelia.lindner@cern.ch
+
+#ifndef ALICEO2_TRACKCUTS_H
+#define ALICEO2_TRACKCUTS_H
+
+#include "Framework/Logger.h"
+#include "Framework/DataTypes.h"
+#include "ReconstructionDataFormats/Track.h"
+#include "DataFormatsGlobalTracking/RecoContainer.h"
+#include "ReconstructionDataFormats/MatchInfoTOF.h"
+#include "DataFormatsTPC/TrackTPC.h"
+#include "DataFormatsTRD/TrackTRD.h"
+#include "GlobalTracking/TrackMethods.h"
+#include "DetectorsBase/Propagator.h"
+#include "ReconstructionDataFormats/VtxTrackIndex.h"
+#include <set>
+#include <vector>
+#include "Rtypes.h"
+#include <gsl/span>
+namespace o2
+{
+using GID = o2::dataformats::GlobalTrackID;
+using GIndex = o2::dataformats::VtxTrackIndex;
+class TrackCuts
+{
+ public:
+  //////////////////////////////// O2 ////////////////////////////////
+  bool isSelected(GID trackIndex, o2::globaltracking::RecoContainer& data)
+  {
+    o2::track::TrackParCov trk;
+    auto contributorsGID = data.getSingleDetectorRefs(trackIndex);
+    auto src = trackIndex.getSource();                       // make selections depending on track source
+    if (contributorsGID[GIndex::Source::ITS].isIndexSet()) { // ITS tracks selection
+      const auto& itsTrk = data.getITSTrack(contributorsGID[GID::ITS]);
+      int ITSnClusters = itsTrk.getNClusters();
+      float ITSchi2 = itsTrk.getChi2();
+      float itsChi2NCl = ITSnClusters != 0 ? ITSchi2 / (float)ITSnClusters : 0;
+      uint8_t itsClusterMap = itsTrk.getPattern();
+      if (itsTrk.getPt() >= mMinPt && itsTrk.getPt() <= mMaxPt &&
+          itsTrk.getEta() >= mMinEta && itsTrk.getEta() <= mMaxEta &&
+          ITSnClusters >= mMinNClustersITS &&
+          itsChi2NCl <= mMaxChi2PerClusterITS &&
+          TrackMethods::FulfillsITSHitRequirements(itsClusterMap, mRequiredITSHits)) {
+        return true;
+      } else {
+        return false;
+      }
+    } else if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
+      const auto& tpcTrk = data.getTPCTrack(contributorsGID[GID::TPC]);
+      uint8_t tpcNClsShared, tpcNClsFound, tpcNClsCrossed, tpcNClsFindable, tpcChi2NCl;
+      // tpcNClsFindable = tpcTrk.getNClusters();
+      // tpcChi2NCl = tpcTrk.getNClusters() ? tpcTrk.getChi2() / tpcTrk.getNClusters() : 0;
+      // o2::TrackMethods::countTPCClusters(tpcTrk, data.getTPCTracksClusterRefs(), data.clusterShMapTPC, data.getTPCClusters(), tpcNClsShared, tpcNClsFound, tpcNClsCrossed);
+      // double tpcCrossedRowsOverFindableCls = tpcNClsCrossed / tpcNClsFindable;
+      math_utils::Point3D<float> v{};
+      std::array<float, 2> dca;
+      if (tpcTrk.getPt() < mPtTPCCut ||
+          std::abs(tpcTrk.getEta()) > mEtaTPCCut ||
+          tpcTrk.getNClusters() < mNTPCClustersCut ||
+          (!(const_cast<o2::tpc::TrackTPC&>(tpcTrk).propagateParamToDCA(v, mBz, &dca, mDCACut)) ||
+           std::abs(dca[0]) > mDCACutY)) {
+        return false;
+      } else {
+        return true;
```

Comment:
again, just the fact that TPC-specific cuts are passed, does not mean that you should validate the global track selection.